vcstool/commands/validate.py

```diff
@@ -0,0 +1,154 @@
+        '--retry', type=int, metavar='N', default=2,
+        help='Retry commands requiring network access N times on failure')
+    return parser
+
+
+def generate_jobs(repos, args):
+    jobs = []
+    for path, repo in repos.items():
+        clients = [c for c in vcstool_clients if c.type == repo['type']]
+        if not clients:
+            from vcstool.clients.none import NoneClient
+            job = {
+                'client': NoneClient(path),
+                'command': None,
+                'cwd': path,
+                'output':
+                    "Repository type '%s' is not supported" % repo['type'],
+                'returncode': NotImplemented
+            }
+            jobs.append(job)
+            continue
+
+        client = clients[0](args.path)
+        args.real_path = path
+        command = ValidateCommand(
+            args, repo['url'],
+            str(repo['version']) if 'version' in repo else None)
+        job = {'client': client, 'command': command}
+        jobs.append(job)
+    return jobs
+
+
+def output_result(result, hide_empty=False):
+    from vcstool.streams import stdout
+    output = result['output']
+    if hide_empty and result['returncode'] is None:
+        output = ''
+    if result['returncode'] == NotImplemented:
+        if output:
+            output = ansi('yellowf') + output + ansi('reset')
+    elif result['returncode']:
+        if not output:
+            output = 'Failed with return code %d' % result['returncode']
+        output = ansi('redf') + output + ansi('reset')
+    elif not result['cmd']:
+        if output:
+            output = ansi('yellowf') + output + ansi('reset')
+    if output or not hide_empty:
+        client = result['client']
+        command = result['command']
+        print(
+            (ansi('redf') if result['returncode'] != NotImplemented and
+                result['returncode'] else ansi('bluef')) +
+            ('INVALID ' if result['returncode'] != NotImplemented and
+                result['returncode'] else 'VALID ') +
+            ansi('bluef') + ansi('boldon') + command.real_path +
+            ansi('boldoff') + ' (' + client.__class__.type + ') ' +
+            ansi('reset'),
+            file=stdout)
+    if (result['command'].debug and result['returncode']) or \
+            result['returncode'] == NotImplemented:
+        if output:
+            try:
+                print(output, file=stdout)
+            except UnicodeEncodeError:
+                print(
+                    output.encode(sys.getdefaultencoding(), 'replace'),
+                    file=stdout)
+
+
+def output_results(results, output_handler=output_result, hide_empty=False):
+    # output results in alphabetic order
+    path_to_idx = {
+        result['command'].real_path: i for i, result in enumerate(results)}
+    idxs_in_order = [path_to_idx[path] for path in sorted(path_to_idx.keys())]
+    for i in idxs_in_order:
+        output_handler(results[i], hide_empty=hide_empty)
```

Comment:
Instead of generating completely custom output reuse the existing output result functions.